src/ucp/core/ucp_context.c

```diff
@@ -674,6 +712,16 @@ ucs_status_t ucp_init_version(unsigned api_major_version, unsigned api_minor_ver
         goto err;
     }
 
+    if (params->thread_mode == UCS_THREAD_MODE_MULTI) {
+        if (config->ctx.use_mt_mutex == 0) {
+            ucs_spinlock_init(&(context->thread_spinlock));
+        } else {
+            pthread_mutex_init(&(context->thread_mutex), NULL);
+        }
+    }
+
+    UCP_CONTEXT_THREAD_CS_ENTER(context);
```

Comment:
no need to take mutex when creating the context. No other thread could access this context anyway before it is created.